client/js/utils/utils.js

```diff
@@ -0,0 +1,32 @@
+var _ = {};
+
+_.each = function(collection, callback){
+  if(collection instanceof Array){
+    for(var i = 0; i < collection.length; i++){
+      callback(collection[i], i, collection);
+    }
+  }else{
+    for(var key in collection){
+      callback(collection[key], key, collection);
+    }
+  }
+};
+
+_.map = function(collection, callback){
+  var result = [];
+  _.each(collection, function(item, key, collection){
+    result.push(callback(item, key, collection));
+  });
+  return result;
+};
+
```

Comment:
will reduce accept objects? or just arrays - i think adding a comment here will make things more clear on what type the collection should be. i am assuming you can pass in both, since each accepts both